restli-client/src/main/java/com/linkedin/restli/client/ActionRequestBuilder.java

```diff
@@ -193,7 +195,10 @@ public ActionRequestBuilder(String baseUriTemplate, TypeSpec<V> elementType, Res
     if (_resourceSpec.getRequestMetadata(_name) == null) // old builder code in use
     {
       requestDataSchema = DynamicRecordMetadata.buildSchema(_name, _actionParams.keySet());
-
+      if (_elementType == null)
+      {
+        _elementType = new TypeSpec<>(_elementClass);
```

Comment:
This is not loading from resourespec?

Reply:
this is old request builder code where spec doesn't have that info. see the else part

Reply:
Ah, didn't notice that.